tests/test_builtin_casters.py

```diff
@@ -256,6 +256,13 @@ class DeepThought(object):
         def __int__(self):
             return 42
 
+    class DoubleThought(object):
```

Comment:
Double meaning it has index and int?

Reply:
Yes, that's what I meant :-) As admitted yesterday:

> yes, this naming joke got out of hand quite quickly

Should I still pick better names?

Reply:
Yes. IntIndexThought, at least. :)

Reply:
Done so. This should make things more clear? More boring as well, but ...

7fd2db5a1105aa073edc46f23e20b063ed7ec04f has _only_ changes to our own tests, so if these pass, look good/better to you, and you're happy with the way `DoubleThought`/`IntAndIndex` is handled, then we can merge this, @henryiii

Reply:
> Yes. IntIndexThought, at least. :)

I liked the mix of HHGG and 1984, but yes ;-)